packages/payment-widget/src/lib/components/payment-confirmation.svelte

```diff
@@ -0,0 +1,418 @@
+<script lang="ts">
+  import CloseIcon from "@requestnetwork/shared-icons/close.svelte";
+  import CopyIcon from "@requestnetwork/shared-icons/copy-icon.svelte";
+  import ExchangeIcon from "@requestnetwork/shared-icons/exchange.svelte";
+  import InfoCircleIcon from "@requestnetwork/shared-icons/info-circle.svelte";
+  import type { Web3Modal } from "@web3modal/ethers5";
+  import { onDestroy, onMount } from "svelte";
+  import type { Currency } from "../types";
+  import { chains } from "../utils/chains";
+  import { NETWORK_LABEL } from "../utils/currencies";
+  import {
+    handleRequestPayment,
+    prepareRequestParameters,
+  } from "../utils/request";
+  import Spinner from "./spinner.svelte";
+  import WalletInfo from "./wallet-info.svelte";
+
+  export let selectedCurrency: Currency;
+  export let amountInUSD: number;
+  export let sellerAddress: string;
+  export let currentPaymentStep: string;
+  export let web3Modal: Web3Modal | null;
+  export let isConnected: boolean;
+  export let persistRequest: boolean;
+  export let onPaymentSuccess: (request: any) => void;
+  export let onPaymentError: (error: string) => void;
+  const COUNTDOWN_INTERVAL = 30;
+
+  let amountInCrypto: number = 0;
+  let countdown: number = COUNTDOWN_INTERVAL;
+  let intervalId: NodeJS.Timeout;
+  let exchangeRate: number = 0;
+  let error: string = "";
+
+  $: isLoadingPrice = true;
```

Comment:
I think the starting value should be `false`, unless that breaks the logic somehow.

```suggestion
  $: isLoadingPrice = false;
```

Reply:
When the modal opens we set it as true in order to disable the pay button, otherwise there is a possibility of having pay button not disabled and clickable before we are able to fetch the price.